### ISSUE TITLE
Support encoding of Transfer-Encoding: chunked

### ISSUE THREAD

**Issue by jfallows** [reaktivity/nukleus-http.java#25]
...in `SourceOutputStream` / `ClientConnectStream`
...and `TargetOutputEstablishedStream` / `ServerAcceptReplyStream`.

**Comment by cmebarrow** [comment]
My chunked.transfer.encoding branch has a preliminary implementation for SourceOutputStream. Tests and code need to be added to handle case where target window is insufficient to write out an incoming data frame because of the extra bytes needed for chunk size line (size+CRLF) and terminator CRLF), or to write the last chunk in processEnd.

**Comment by jfallows** [comment]
@cmebarrow is `Transfer-Encoding: chunked` supported by both HTTP/1.1 responses _and_ HTTP/1.1 requests?

**Comment by cmebarrow** [comment]
Yes it can be used in both. This is implied for example in https://tools.ietf.org/html/rfc7230#section-3.3.3.
